reana/reana_dev/git.py

```diff
@@ -82,6 +85,45 @@ def get_current_commit(srcdir):
     )
 
 
+def git_is_current_version_tagged(component):
+    """Determine whether the current version in source code is present as a git tag."""
+    tags_history = run_command(
+        "git tag --list", component, display=False, return_output=True
+    )
+    current_version = get_component_current_version_from_file(component)
+    return current_version in tags_history
+
+
+def git_create_release_commit(component):
+    """Create a release commit for the given component."""
+    if "release: v" in get_current_commit(get_srcdir(component)):
+        display_message("Nothing to do, last commit is a release commit.", component)
+        return False
+
+    if not git_is_current_version_tagged(component):
+        display_message(
+            f"Current version ({get_component_current_version_from_file(component)}) "
+            "not present as a git tag, plese release it and add a tag.",
+            component,
+        )
+        return False
+
+    next_version = bump_component_version(
+        component, get_component_current_version_from_file(component),
+    )
+
+    if (
+        run_command(
+            "git branch --show-current", component, display=False, return_output=True,
+        )
+        == "master"
+    ):
+        run_command(f"git checkout -b release-{next_version}", component)
+
+    run_command(f"git commit -m 'release: {next_version}'", component)
```

Comment:
You need to stage modified files first? Two options:

- use wild staging: `git commit -a -m ...` could do the job, but it might be dangerous
- use targeted staging: `git add <version_file>` to add only the version file you incremented (python or npm)